libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -529,71 +531,15 @@ internal class MapboxTripSession(
     }
 
     private fun updateRawLocation(rawLocation: Location) {
-        unconditionalStatusPollingJob?.cancel()
         if (state != TripSessionState.STARTED) return
 
         this.rawLocation = rawLocation
         locationObservers.forEach { it.onRawLocationChanged(rawLocation) }
         mainJobController.scope.launch {
             navigator.updateLocation(rawLocation.toFixLocation())
-            updateDataFromNavigatorStatus()
-        }
-
-        unconditionalStatusPollingJob = ioJobController.scope.launch {
-            delay(UNCONDITIONAL_STATUS_POLLING_PATIENCE)
-            while (isActive) {
-                mainJobController.scope.launch {
-                    updateDataFromNavigatorStatus()
-                }
-                delay(UNCONDITIONAL_STATUS_POLLING_INTERVAL)
-            }
         }
     }
 
-    private fun updateDataFromNavigatorStatus() {
-        val updateNavigatorStatusDataJob = mainJobController.scope.launch {
-            if (state != TripSessionState.STARTED) {
-                return@launch
-            }
-
-            val status = getNavigatorStatus()
-            if (!isActive) {
-                return@launch
-            }
-            val enhancedLocation = status.navigationStatus.location.toLocation()
-            val keyPoints = status.navigationStatus.keyPoints.toLocations()
-            updateEnhancedLocation(enhancedLocation, keyPoints)
-            if (!isActive) {
-                return@launch
-            }
-            updateMapMatcherResult(status.getMapMatcherResult(enhancedLocation, keyPoints))
-            if (!isActive) {
-                return@launch
-            }
-            val remainingWaypoints = ifNonNull(status.route?.routeOptions()?.coordinates()?.size) {
-                it - status.navigationStatus.nextWaypointIndex
-            } ?: 0
-            val routeProgress = getRouteProgressFrom(
-                status.route,
-                status.navigationStatus,
-                remainingWaypoints
-            )
-            updateRouteProgress(routeProgress)
-            if (!isActive) {
-                return@launch
-            }
-            isOffRoute = status.navigationStatus.routeState == RouteState.OFF_ROUTE
-        }
-        updateNavigatorStatusDataJob.invokeOnCompletion {
-            updateNavigatorStatusDataJobs.remove(updateNavigatorStatusDataJob)
-        }
-        updateNavigatorStatusDataJobs.add(updateNavigatorStatusDataJob)
-    }
-
-    private suspend fun getNavigatorStatus(): TripStatus {
-        return navigator.getStatus(navigationOptions.navigatorPredictionMillis)
```

Comment:
As mentioned above, the `navigationOptions.navigatorPredictionMillis` should still be respected and public.